### PR TITLE
Include implementations for new ItemMeta types and update as required.

### PR DESCRIPTION
**Pull request by nickrobson**
This adds data-types for meta belonging to:

* Fireworks
* Skulls
* Maps
* Enchanted Books (including Enchantments)

Matching PR on SDK: TridentSDK/TridentSDK#50

### CODE REVIEW

**Comment by nickrobson** [comment]
Build fails due to not being built against my PR on TridentSDK.

**Comment by mkotb** [comment]
Please remove the lines at the start of class definition and above the end bracket. Additionally, some of the added classes do not contain the LICENSE header. Add them by running `mvn clean`

File: src/main/java/net/tridentsdk/server/data/item/EnchantedBookMetaImpl.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    @NBTField(name = "StoredEnchantments", type = TagType.LIST)
+    List<NBTEnchantment> enchants;
+
+    @NBTField(name = "RepairCost", type = TagType.STRING)
+    int repairCost = 0;
+
+    Map<Enchantment, Short> enchantments = Maps.newConcurrentMap();
+
+    public void process() {
```
Comment (mkotb) [inline src/main/java/net/tridentsdk/server/data/item/EnchantedBookMetaImpl.java]: Why is there no `@Override` here?

File: src/main/java/net/tridentsdk/server/data/item/SkullMetaImpl.java [hunk 2]
```diff
@@ -16,12 +16,25 @@
...
 import net.tridentsdk.meta.item.SkullMeta;
+import net.tridentsdk.meta.item.SkullTexture;
 import net.tridentsdk.meta.nbt.NBTField;
 import net.tridentsdk.meta.nbt.NBTSerializable;
 import net.tridentsdk.meta.nbt.TagType;
 
 public class SkullMetaImpl extends ItemMetaImpl implements SkullMeta {
+	
+	protected static final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
```
Comment (mkotb) [inline src/main/java/net/tridentsdk/server/data/item/SkullMetaImpl.java]: Switch the Tabs to spaces

File: src/main/java/net/tridentsdk/server/data/item/SkullMetaImpl.java [hunk 3]
```diff
@@ -38,10 +51,96 @@ public void setOwner(String name) {
...
+    	String signature;
+    	
+    	String jsonString;
+    	JsonObject json;
+    	
+        long jsonTimestamp;
+        String jsonProfileId, jsonProfileName;
+        boolean jsonIsPublic;
+        String jsonSkinUrl, jsonCapeUrl;
+
```
Comment (mkotb) [inline src/main/java/net/tridentsdk/server/data/item/SkullMetaImpl.java]: Indentation is not aligned

File: src/main/java/net/tridentsdk/server/data/item/ExplosionMetaImpl.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+import com.google.common.collect.Lists;
+
+import net.tridentsdk.base.Color;
+import net.tridentsdk.meta.item.ExplosionMeta;
+import net.tridentsdk.meta.nbt.NBTField;
+import net.tridentsdk.meta.nbt.NBTSerializable;
+import net.tridentsdk.meta.nbt.TagType;
+
+public class ExplosionMetaImpl implements ExplosionMeta, NBTSerializable {
```
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/server/data/item/ExplosionMetaImpl.java]: Please lower the visibility modifiers and encapsulate as necessary.

File: src/main/java/net/tridentsdk/server/data/item/SkullMetaImpl.java [hunk 5]
```diff
@@ -38,10 +51,96 @@ public void setOwner(String name) {
...
+    	@NBTField(name = "Signature", type = TagType.STRING)
+    	String signature;
+    	
+    	String jsonString;
+    	JsonObject json;
+    	
+        long jsonTimestamp;
+        String jsonProfileId, jsonProfileName;
+        boolean jsonIsPublic;
+        String jsonSkinUrl, jsonCapeUrl;
```
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/server/data/item/SkullMetaImpl.java]: Please avoid declaring more than one field per line.

File: src/main/java/net/tridentsdk/server/data/item/ExplosionMetaImpl.java (hunk 4 above)
Comment (nickrobson) [inline src/main/java/net/tridentsdk/server/data/item/ExplosionMetaImpl.java]: Please define "visibility modifier". If you mean private, protected, public, etc, I think it's fine.

Comment (AgentTroll) [inline src/main/java/net/tridentsdk/server/data/item/ExplosionMetaImpl.java]: Unless absolutely necessary, `private` field visibility is required.

Comment (nickrobson) [inline src/main/java/net/tridentsdk/server/data/item/ExplosionMetaImpl.java]: Okay. You just commented on the class definition which is why I was confused.

Comment (AgentTroll) [inline src/main/java/net/tridentsdk/server/data/item/ExplosionMetaImpl.java]: I did so to indicate the entire class (and entire PR needs to be considered)

Comment (nickrobson) [inline src/main/java/net/tridentsdk/server/data/item/ExplosionMetaImpl.java]: Also, I hope you realise that these fields are package-level not public lol.

**Comment by nickrobson** [comment]
@mkotb @AgentTroll I believe everything is resolved here.

File: src/main/java/net/tridentsdk/server/data/item/ExplosionMetaImpl.java (hunk 4 above)
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/server/data/item/ExplosionMetaImpl.java]: I understand. Please set to private there is no other way.

Comment (nickrobson) [inline src/main/java/net/tridentsdk/server/data/item/ExplosionMetaImpl.java]: I did this anyway.

**Comment by AgentTroll** [comment]
Thank you for your contributions.

**Comment by nickrobson** [comment]
:+1:

**Pull request merged**